### PR TITLE
Preprocess pipeline yaml documentation

### PR DESCRIPTION
**Pull request by Lilly-May**
This PR introduces a separate documentation file for all the preprocess `pipeline.yml` parameters. Additionally, I removed all the parameter explanations from the `pipeline.yml` itself to make the file easier to read.

Once this PR is merged, meaning we have settled on the final format of the file, I'll also adjust the prefilled file for the preprocessing tutorial to match the structure here.

There were a few things I wasn't entirely sure about, and I'll mention those in separate comments below.

### CODE REVIEW

File: docs/yaml_docs/pipeline_preprocess_yml.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+
+    - <span class="parameter">n_cells_by_counts</span> `Integer`<br>
+
+  - <span class="parameter">max</span><br>
+
+    - <span class="parameter">total_counts</span> `Integer`<br>
+
+    - <span class="parameter">n_cells_by_counts</span> `Integer`<br>
+
+### Protein-specific filtering (prot)
```
Comment (Lilly-May) [inline docs/yaml_docs/pipeline_preprocess_yml.md:169]: Should we still list the entire structure (obs: min / max, var: min / max) here, although there are no specific parameters in the respective sections?

File: docs/yaml_docs/pipeline_preprocess_yml.md [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+  - <span class="parameter">n_comps</span> `Integer`, Default: 50<br>
+        Number of components to compute.
+
+  - <span class="parameter">solver</span> `String`, Default: default<br>
+        If using PCA, which solver to use. Setting this parameter to "default", will use the 'arpack' solver.
+
+  - <span class="parameter">color_by</span> `String`, Default: sample_id<br>
+        Specify the covariate you want to use to color the dimensionality reduction plot.
+
+  - <span class="parameter">dim_remove</span> `TODO`<br>
```
Comment (Lilly-May) [inline docs/yaml_docs/pipeline_preprocess_yml.md:426]: I'm not sure in what format the `dim_remove` parameter should be provided. As a boolean? Or an integer?

File: panpipes/panpipes/pipeline_preprocess/pipeline.yml [hunk 3]
```diff
@@ -1,328 +1,203 @@
...
   threads_high: 2
-  # this must be enough memory to load your mudata and do computationally light tasks
   threads_medium: 2
-  # this must be enough memory to load text files and do plotting, requires much less memory than the other two
   threads_low: 1
-# path to conda env, leave blank if running native or your cluster automatically inherits the login node environment
+
 condaenv:
 
 # allows for tweaking where the jobs get submitted, in case there is a special queue for long jobs or you have access to a gpu
```
Comment (Lilly-May) [inline panpipes/panpipes/pipeline_preprocess/pipeline.yml:18]: What does this comment refer to? The `condaenv`? If yes, I don't think I understand what is meant by that.

Comment (bio-la) [reply panpipes/panpipes/pipeline_preprocess/pipeline.yml:18]: yes this refers to the path to the conda environment if you're running this workflow with a specific  conda env. 

notice you can install panpipes in a main conda env  (/path/to/panpipes_env/) and have a separate conda environment JUST for this workflow (path/to/preprocess_env).

running panpipes after activating the conda environment /path/to/panpipes_env/ should ensure that the environment is inherited when running the pipeline if you are on an interactive session (or on a PC) 

when jobs are spawn, we found that different clusters behave differently so we always leave the option to specify the conda env (whether it's /path/to/panpipes_env/ or /path/to/preprocess_env/) in the yml

there is a section about this in the installation, but maybe not superclear?
https://panpipes-pipelines.readthedocs.io/en/latest/install.html#specifying-conda-environments-to-run-panpipes

**Review by bio-la — COMMENTED** [review]
can you add the .md to the index so it's rendered in the documentation and i can check how the filtering dictionary is rendered (and answer your comment)?
thanks!

File: panpipes/panpipes/pipeline_preprocess/pipeline.yml (hunk 3 above)
Comment (Lilly-May) [reply panpipes/panpipes/pipeline_preprocess/pipeline.yml]: Ah, I see - thank you for the explanation! I deleted that part from the `pipeline.yml` file itself but added the link you send here to the parameter documentation. I think that should clarify things.

File: docs/yaml_docs/pipeline_preprocess_yml.md [hunk 4]
```diff
@@ -328,8 +329,8 @@ Whether applying scaling or not is still a matter of debate, as stated in the [L
         Parameter for CLR normalisation.
         The CLR margin determines whether you normalise per cell (as you would normalise RNA data), or by feature (recommended, due to the variable nature of protein assays). 
         Hence, CLR margin 0 is recommended for informative qc plots in this pipeline.
```
Comment (bio-la) [inline docs/yaml_docs/pipeline_preprocess_yml.md:331]: margin =1 is recommended

**Review by bio-la — CHANGES_REQUESTED** [review]

**Review by bio-la — APPROVED** [review]

**Pull request merged**